apps/segment/src/pages/api/webhooks/order-cancelled.ts

```diff
@@ -27,31 +29,46 @@ const handler: NextWebhookApiHandler<OrderUpdatedSubscriptionPayloadFragment> =
   const { authData, payload } = context;
 
   if (!payload.order) {
-    return res.status(200).end();
+    logger.info("Payload does not contain order data. Skipping.");
+    return res
+      .status(200)
+      .json({ message: "Payload does not contain order data. It will be skipped by app" });
   }
 
+  loggerContext.set(ObservabilityAttributes.ORDER_ID, payload.order.id);
+
   try {
     const segmentEventTracker = await createSegmentClientForWebhookContext({ authData });
 
-    logger.info("Sending order cancelled event to Segment");
-
     await segmentEventTracker.trackEvent(
       trackingEventFactory.createOrderCancelledEvent(payload.order),
     );
 
-    return res.status(200).end();
+    logger.info("Order cancelled event successfully sent to Segment");
+
+    return res.status(200).json({ message: "Order cancelled event successfully sent to Segment" });
   } catch (e) {
-    if (e instanceof SegmentNotConfiguredError) {
+    if (e instanceof SegmentWriteKeyNotFoundError) {
       // todo disable webhooks if not configured
 
-      return res.status(200).end();
+      logger.warn(
+        "Segment write key is not set in app configuration. Contact client to fix the issue.",
+      );
```

Comment:
question: do we want to treat broken app config as warning for our logs? I think this is fine for now, but imho it's not really scalable and client logs would be better ;)

Reply:
In other PRs I disable webhooks is there is no key so hopefully there won't be many such situations but I agree - we should let client know (client logs or maybe emails etc.)